saleor/discount/utils.py

```diff
@@ -186,6 +187,81 @@ def get_voucher_code_instance(
     return code_instance
 
 
+def apply_voucher_to_line(
+    voucher_info: "VoucherInfo",
+    lines_info: Iterable["LineInfo"],
+):
+    """Attach voucher to valid checkout or order lines info.
+
+    Apply a voucher to checkout/order line info when the voucher has the type
+    SPECIFIC_PRODUCTS or is applied only to the cheapest item.
+    """
+    from .utils import get_discounted_lines
+
+    voucher = voucher_info.voucher
+    discounted_lines_by_voucher: list["LineInfo"] = []
+    lines_included_in_discount = lines_info
+    if voucher.type == VoucherType.SPECIFIC_PRODUCT:
+        discounted_lines_by_voucher.extend(
+            get_discounted_lines(lines_info, voucher_info)
+        )
+        lines_included_in_discount = discounted_lines_by_voucher
+    if voucher.apply_once_per_order:
+        cheapest_line = _get_the_cheapest_line(lines_included_in_discount)
+        if cheapest_line:
+            discounted_lines_by_voucher = [cheapest_line]
+    for line_info in lines_info:
+        if line_info in discounted_lines_by_voucher:
+            line_info.voucher = voucher
+            line_info.voucher_code = voucher_info.voucher_code
+
+
+def get_discounted_lines(
```

Comment:
Correct me if I am wrong. `get_discounted_lines` is a replacement for custom `get_discounted_lines` functions used with Checkout, and Order (`saleor.checkout.utils.get_discounted_lines`, `saleor.order.utils.get_discounted_lines`). Shouldn't we fully drop these functions?